heft-plugins/heft-lint-plugin/src/Eslint.ts

```diff
@@ -100,12 +102,15 @@ export class Eslint extends LinterBase<TEslint.ESLint.LintResult> {
       };
     }
 
+    process.env.ESLINT_BULK_ESLINTRC_FOLDER_PATH = path.dirname(linterConfigFilePath);
     this._linter = new eslintPackage.ESLint({
       cwd: buildFolderPath,
       overrideConfigFile: linterConfigFilePath,
       // Override config takes precedence over overrideConfigFile
       overrideConfig,
-      fix: fixFn
+      fix: fixFn,
+      // If requested, disable the scan for .eslintrc files relative to linted files
+      useEslintrc: !disableLintConfigSearch
```

Comment:
I don't understand this option. According to the ESLint docs, it seems to completely suppress configuration:

> **To Address:** You should be sure to use a configuration whenever you run ESLint. However, you can still run ESLint without a configuration by passing the `--no-eslintrc` option on the command line or setting the `useEslintrc` option to `false` for `CLIEngine`.

...but then where does the configuration come from?  Is the idea that Heft itself locates `.eslintrc.js` and then provides its path via `overrideConfigFile`?

If so, then does Heft actually support multiple other files (probed using `useEslintrc`)?  How would that interact with `overrideConfigFile`?

Without a test project verifying/illustrating how this would work, it might be better to set `useEslintrc=false` by default (or hardwire it to always be false).  My concern is that quite a lot of people may upgrade Heft without realizing that they're supposed to update their rigs to set `disableLintConfigSearch=true`, and then they'll permanently be invoking `heft-lint-plugin` in a nonstandard configuration.

Reply:
Well, the _current_ release version of `heft-lint-plugin` leaves `useEslintrc` as the default value of `true`, which causes a ancestor file scan from the file being linted until it finds any of the valid eslintrc files, so changing the value of this option constitutes a breaking change.

Note that `heft-lint-plugin` only enables ESLint if there exists a `.eslintrc.cjs` (or a couple other names) in the project root, and it takes the path to that file and passes it into ESLint via the `overrideConfigFile` option.

In any sane file layout, the search will terminate at the root .eslintrc.cjs file, but the scanning makes it possible to have customized configurations in subfolders. Now, neither of the repositories I work with _have_ nested eslintrc files, so the only observable difference in said repositories is a significant reduction in the number of syscalls issued by ESLint.

If a project has an eslintrc file in a subfolder of the project root, however, changing this flag will cause that file to be ignored, which is likely unexpected (otherwise, why does such a file exist?).

Reply:
https://github.com/eslint/eslintrc/blob/f05aea7eeca2b99eae6310954cb25d3a904cc605/lib/cascading-config-array-factory.js#L351-L361